source/bson-corpus/bson-corpus.rst

```diff
@@ -314,21 +305,48 @@ manner.
 Testing parsing errors
 ----------------------
 
-The interpretation of ``parseErrors`` is type-specific.  For example,
-helpers for creating Decimal128 values may parse strings to convert them
-to binary Decimal128 values.  The ``parseErrors`` cases are strings that
-will *not* convert correctly.
+The interpretation of ``parseErrors`` is type-specific. The structure of test
+cases within ``parseErrors`` is described in `Parse error case keys`_.
 
-The documentation for a type (if any) will specify how to use these
-cases for testing.
+Drivers SHOULD test that each case results in a parsing error (e.g. parsing
+Extended JSON, constructing a language type). Implementations MAY test
+assertions in an implementation-specific manner.
 
-For type "0x00" (i.e. top-level documents), the ``parseErrors`` entries have a
-``description`` field and an ``string`` field.  Parsing the ``string`` field
-as Extended JSON MUST result in an error.
 
-Drivers SHOULD test that each case results in a parse error.
-Implementations MAY test assertions in an implementation-specific
-manner.
+Top-level Document (type 0x00)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For type "0x00" (i.e. top-level documents), the ``string`` field contains input
+for an Extended JSON parser. Drivers MUST parse the Extended JSON input using an
+Extended JSON parser and verify that doing so yields an Extended JSON parsing
+error.
+
+Drivers MUST also parse the Extended JSON input using a regular JSON parser (not
```

Comment:
```suggestion
Drivers SHOULD also parse the Extended JSON input using a regular JSON parser (not
```

C does not test with a regular JSON parser. It seems odd to me to require that drivers test the test is written correctly.

Reply:
Nor does Java.  I would prefer not to introduce a 3rd party JSON parser (The JDK for example, doesn't provide one) just for this.

Reply:
Changed to SHOULD per Kevin's suggestion.